### ISSUE TITLE
Many false positives for Uild.Tostring

### ISSUE THREAD

**Issue by ChristianSauer** [meziantou/Meziantou.Analyzer#921]
### Version of the Meziantou.Analyzer NuGet package

2.0.238

### Rule Identifier

MA0011

### Target Framework

net8.0

### C# Language version

C# 12

### Description

We are using this package:
https://github.com/Cysharp/Ulid

It provides a ToString Method and even the usual overloads but de facto all Methods are Culture independent.
So raising MA0011 seems useless

For the implementation of ToString see: https://github.com/Cysharp/Ulid/blob/9594ed65e737977a22e7835c8569d33a9acb1a25/src/Ulid/Ulid.cs#L514

Example 

### Reproduction Steps

Minimal code:

```c#
Ulid.NewUlid().ToString();
```

**Comment by drieseng** [comment]
@ChristianSauer, you can define an assembly-level **CultureInsensitiveTypeAttribute** for each type in that package that you want to mark as being culture invariant.

For example:
```c#
[assembly:CultureInsensitiveType(typeof(Ulid))]
```

**Comment by ChristianSauer** [comment]
@drieseng TIL 👍🏻  I like that.
Unfortunately, we use Ulid in multiple assemblies. But still MUCH better than commenting the call everywhere.

**Comment by meziantou** [comment]
I'm updating the documentation to mention this attribute (in fact, Copilot is doing it). 

To add the attribute everywhere, you can create a `.cs` file containing the attributes, and then add it to all projects using the `Directory.Build.{props,targets}` file. Or you can create a CodingStandard or sdk package like I do in all my project that would contain these attributes. More info:
- https://github.com/meziantou/Meziantou.DotNet.CodingStandard / https://www.meziantou.net/sharing-coding-style-and-roslyn-analyzers-across-projects.htm
- https://github.com/meziantou/Meziantou.NET.Sdk
